frontend/cs/scs/api_assertions.go

```diff
@@ -128,6 +128,33 @@ func (builder *builder) AssertIsBoolean(i1 frontend.Variable) {
 
 }
 
+func (builder *builder) AssertIsCrumb(i1 frontend.Variable) {
+	const errorMsg = "AssertIsCrumb: input is not a crumb"
+	if c, ok := builder.constantValue(i1); ok {
+		bytes := c.Bytes()
+		for i := 1; i < len(bytes); i++ {
+			if bytes[i] != 0 {
+				panic(errorMsg)
+			}
+		}
+		for i := 0; i < 4; i++ {
+			if bytes[0] == byte(i) {
+				return
+			}
+		}
+		panic(errorMsg)
+	}
+
+	x := builder.MulAcc(builder.Mul(-3, i1), i1, i1).(expr.Term)
```

Comment:
can you comment a bit the logic? i.e. reading the code it's not obvious what is the constraint / if it's sound

Reply:
+ also interested to know the justification of this API (i.e clearly say; using the `std/math/bits` package generates X constraints while this generates Y constraint for R1CS, Z for SCS)